### ISSUE TITLE
Missing packages even though texlive-full is installed

### ISSUE THREAD

**Issue by maurerle** [smhaller/ldap-overleaf-sl#47]
Thank you very much for this add-on.

However I still had to add packages with tlmgr after building the docker images using `make build`.
I don't really know why, as this should have a texlive full install already (as specified in the Dockerfile of this repo through apt install texlive-full), but it kept saying "biblatex missing", `xcolor.sty not found` and others.

So I ended up with another Dockerfile doing:

```Dockerfile
FROM ldap-overleaf-sl:latest
RUN tlmgr update --self
RUN tlmgr install scheme-full
```

which still installs 4526 packages.
Before that, the docker image already had a size of 6.6GB

I already looked here:
https://github.com/overleaf/overleaf/wiki/Quick-Start-Guide#latex-environment

I also found out that this is currently using texlive 2019 - is there a way to install a more recent texlive?

**Comment by yzx9** [comment]
Both `texlive` and `texlive-full` are meta-packages, meaning they do not contain any TeX files themselves; therefore, you will need to install these files manually. After installation, the size of the Docker image will increase to 12GB.

As highlighted in the [OverLeaf Release Note 4.0.0](https://github.com/overleaf/overleaf/wiki/Release-Notes--4.x.x#other-changes-3), TeXLive 2023 is now the default version for instances not running Sandboxed Compiles. It is advisable to consider upgrading both our add-on and Overleaf to leverage the latest enhancements.

**Comment by maurerle** [comment]
I don't know, i still get errors when compiling with a bib file `Error: Found biblatex control file version 3.10, expected version 3.7.` because biber is the wrong version (2.14)
https://packages.ubuntu.com/focal/biber
while biblatex is 3.19 from texlive-2023..

I tried installing packages using the above mentioned tlmgr hack on top of the texlive install, as well as without the `apt install texlive` in the dockerfile of this repository.
Very weird situation.

**Comment by smhaller** [comment]
What I do in my environment: I do not install the Debian packages with `apt` but I adapted (removed line 26/27) the [Dockerfile](https://github.com/smhaller/ldap-overleaf-sl/blob/master/ldap-overleaf-sl/Dockerfile)  to install the base already via `tlmgr`
```Dockerfile
RUN apt-get -y install python3-pip libxml-libxslt-perl cpanminus libbtparse2 && \
    pip3 install pygments
RUN tlmgr update --self --all  && \
    tlmgr option repository https://mirror.kumi.systems/ctan/systems/texlive/tlnet && \
    tlmgr install scheme-full --verify-repo=none 
```

On a previous installation I had also issues with biber and back than i solved it by installing it from source using following commands (back than using biber v2.18)
```Dockerfile
RUN cpanm install Module::Build && cpanm install PAR::Packer; rm -fr root/.cpanm; exit 0
RUN cd /home && wget https://github.com/plk/biber/archive/refs/tags/v2.18.zip &&  \
    unzip v2.18.zip && cd biber-2.18/ && perl ./Build.PL && ./Build installdeps && \
    ldconfig && ./Build test && ./Build install
```


EDIT: current Biber Version is 2.19

**Comment by maurerle** [comment]
Using the first approach kind of worked, but as biber was not found in the PATH, I had to find it (using `find / -name biber`) and symlink it:
`RUN ln -s /usr/local/texlive/2023/bin/x86_64-linux/biber biber`

While the image for approach 1 was building, I also tried the second appraoch with `cpanm`.
I needed to also install `libperl-dev` and `cpanminus` to get the second approach to work.
In general the second approach was much slower, as the build of bibers `installdeps` is quite resource intensive (at least on my machine).

Thank you very much!
